pkg/tests/cross-tests/input_cross_test.go

```diff
@@ -217,3 +217,203 @@ func TestInputsEmptyConfigModeAttrSet(t *testing.T) {
 		SkipCompareRawConfig: true,
 	})
 }
+
+func TestOptionalSetNotSpecified(t *testing.T) {
+	skipUnlessLinux(t)
+	runCreateInputCheck(t, inputTestCase{
+		Resource: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"f0": {
+					Optional: true,
+					Type:     schema.TypeSet,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"x": {Optional: true, Type: schema.TypeString},
+						},
+					},
+				},
+			},
+		},
+		Config:               tftypes.NewValue(tftypes.Object{}, map[string]tftypes.Value{}),
+		SkipCompareRawConfig: true,
+	})
+}
+
+func TestInputsEmptyCollections(t *testing.T) {
+	skipUnlessLinux(t)
+	// TODO[pulumi/pulumi-terraform-bridge#1971]
+	t.Skipf("We don't handle missing collections correctly")
+	config := tftypes.NewValue(tftypes.Object{}, map[string]tftypes.Value{})
+
+	// signifies a block
+	resourceElem := &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"x": {Optional: true, Type: schema.TypeString},
+		},
+	}
+
+	// signifies an attribute
+	schemaElem := &schema.Schema{
+		Type: schema.TypeMap,
+	}
+
+	for _, tc := range []struct {
+		name       string
+		maxItems   int
+		typ        schema.ValueType
+		elem       any
+		configMode schema.SchemaConfigMode
+	}{
+		{"list block", 0, schema.TypeList, resourceElem, schema.SchemaConfigModeAuto},
+		{"set block", 0, schema.TypeSet, resourceElem, schema.SchemaConfigModeAuto},
+		// This isn't quite valid but should work
+		{"map block", 0, schema.TypeMap, resourceElem, schema.SchemaConfigModeAuto},
+		{"list max items one block", 1, schema.TypeList, resourceElem, schema.SchemaConfigModeAuto},
+		{"set max items one block", 1, schema.TypeSet, resourceElem, schema.SchemaConfigModeAuto},
+		// This isn't quite valid but should work
+		{"map max items one block", 1, schema.TypeMap, resourceElem, schema.SchemaConfigModeAuto},
+		{"list attr", 0, schema.TypeList, schemaElem, schema.SchemaConfigModeAuto},
+		{"set attr", 0, schema.TypeSet, schemaElem, schema.SchemaConfigModeAuto},
+		{"map attr", 0, schema.TypeMap, schemaElem, schema.SchemaConfigModeAuto},
+		{"list max items one attr", 1, schema.TypeList, schemaElem, schema.SchemaConfigModeAuto},
+		{"set max items one attr", 1, schema.TypeSet, schemaElem, schema.SchemaConfigModeAuto},
+		{"map max items one attr", 1, schema.TypeMap, schemaElem, schema.SchemaConfigModeAuto},
+		{"list config mode attr", 0, schema.TypeList, resourceElem, schema.SchemaConfigModeAttr},
```

Comment:
This is good. I suspect `schema.SchemaConfigModeAttr` should change how the value is processed (and even printed) in TF. It should start using the attr syntax instead of block syntax.